### PR TITLE
Add Subsample to Infer strandedness

### PR DESCRIPTION
**Pull request by robsyme**
EDIT: updated the description + made some changes on the code to reflect what the subworkflow is actually doing (+ some comments on how we actually use it for)

## Overview

Adds a subworkflow to subsample reads which can be used to infer pair strandedness

## PR checklist

Closes #XXX 

- [X] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

**Review by maxulysse — APPROVED** [review]
Love it <3

A few questions though.

- Can a few comments be added about how we can infer it from salmon?
    - (like maybe add that in meta.yml about the json)

File: subworkflows/nf-core/fastq_subsample_fq_salmon/meta.yml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  - json_info:
+    type: file
+    description: |
+      File containing meta information from Salmon quant
+      Which could be used to infer strandedness among other things
+    pattern: "*info.json"
+  - versions:
+    type: file
+    description: File containing software versions
+    pattern: "versions.yml"
```
Comment (nvnieuwk) [inline subworkflows/nf-core/fastq_subsample_fq_salmon/meta.yml:52]:
```suggestion
  - reads:
      type: file
      description: Subsampled fastq reads.
      pattern: "*.{fq,fastq}{,.gz}"
  - results:
      type: directory
      description: Folder containing the quantification results for a specific sample
      pattern: "${prefix}"
  - json_info:
      type: file
      description: |
        File containing meta information from Salmon quant
        Which could be used to infer strandedness among other things
      pattern: "*info.json"
  - versions:
      type: file
      description: File containing software versions
      pattern: "versions.yml"
```

**Review by nvnieuwk — COMMENTED** [review]
A small comment, overall looks good to me from a structure standpoint, but I won't approve since my knowledge of these tools is very limited and I should probably leave this to people with more knowledge

**Comment by maxulysse** [comment]
> A small comment, overall looks good to me from a structure standpoint, but I won't approve since my knowledge of these tools is very limited and I should probably leave this to people with more knowledge

Basically using FQ to subsample the input FASTQ and then running salmon on it.
It can be use to infer the strandedness of the library among other things

**Review by nvnieuwk — APPROVED** [review]
LGTM!

**Comment by robsyme** [comment]
Thanks all!

**Comment by maxulysse** [comment]
Sorry we hijacked your PR

**Pull request merged**